black.py

```diff
@@ -1349,7 +1388,7 @@ def whitespace(leaf: Leaf) -> str:  # noqa C901
 
         elif prevp.type == token.COLON:
             if prevp.parent and prevp.parent.type in {syms.subscript, syms.sliceop}:
-                return NO
+                return SPACE if complex_subscript else NO
```

Comment:
This needs to return `SPACE` because otherwise control would flow to this block of `if`s https://github.com/ambv/black/blob/8f6b5048fcb71bc59399fd7b5511d6e229033b4c/black.py#L1414

Unless you prefer to handle this case down there too